pkg/redis/client.go

```diff
@@ -496,6 +642,28 @@ func (c *redisClient) ZRevRange(ctx context.Context, key string, start int64, st
 	return cmd.(*baseRedis.StringSliceCmd).Val(), err
 }
 
+func (c *redisClient) ZRevRangeWithScores(ctx context.Context, key string, start int64, stop int64) ([]Z, error) {
+	cmd, err := c.execute(ctx, func() ErrCmder {
+		return c.base.ZRevRangeWithScores(ctx, key, start, stop)
+	})
+
+	var result []Z
+	if err == nil {
+		zs := cmd.(*baseRedis.ZSliceCmd).Val()
+		result = make([]Z, len(zs))
+		for i := range zs {
+			result[i] = Z{
+				Score:  zs[i].Score,
+				Member: zs[i].Member,
+			}
+		}
+	} else {
+		result = make([]Z, 0)
+	}
```

Comment:
Please follow this pattern (early return) when coding in gosoline:
```suggestion
	if err != nil {
		return make([]Z, 0), err
	}
	
	zs := cmd.(*baseRedis.ZSliceCmd).Val()
	result = make([]Z, len(zs))
	for i := range zs {
		result[i] = Z{
			Score:  zs[i].Score,
			Member: zs[i].Member,
		}
	}

```

Reply:
Ended up scrapping the error handling. Worst case go-redis returns a nil-slice here, which will then result in a []Z slice of length 0 being returned.